Source/player.cpp

```diff
@@ -1918,199 +1918,112 @@ void __fastcall PlrDeadItem(int pnum, struct ItemStruct *itm, int xx, int yy)
+	for ( i = 0; i < MAXBELTITEMS && hGold > 0; i++ ) {
+		if ( plr[pnum].SpdList[i]._itype == ITYPE_GOLD && plr[pnum].SpdList[i]._ivalue != 5000 ) {
+			if ( hGold < plr[pnum].SpdList[i]._ivalue ) {
+				plr[pnum].SpdList[i]._ivalue -= hGold;
+				SetSpdbarGoldCurs(pnum, i);
+				SetPlrHandItem(&plr[pnum].HoldItem, IDI_GOLD);
+				GetGoldSeed(pnum, &plr[pnum].HoldItem);
+				SetPlrHandGoldCurs(&plr[pnum].HoldItem);
+				plr[pnum].HoldItem._ivalue = hGold;
+				PlrDeadItem(pnum, &plr[pnum].HoldItem, 0, 0);
+				hGold = 0;
+			} else {
+				hGold -= plr[pnum].SpdList[i]._ivalue;
+				RemoveSpdBarItem(pnum, i);
+				SetPlrHandItem(&plr[pnum].HoldItem, IDI_GOLD);
+				GetGoldSeed(pnum, &plr[pnum].HoldItem);
+				SetPlrHandGoldCurs(&plr[pnum].HoldItem);
+				plr[pnum].HoldItem._ivalue = plr[pnum].SpdList[i]._ivalue;
+				PlrDeadItem(pnum, &plr[pnum].HoldItem, 0, 0);
+				i = -1;
 			}
 		}
-		if ( ++i >= 8 )
-		{
-			if ( v3 > 0 )
-			{
-				ia = 0;
-				do
-				{
-					if ( v3 <= 0 )
-						break;
-					v6 = 368 * ia + v2 * 21720;
-					v15 = v6;
-					if ( *(int *)((char *)&plr[0].SpdList[0]._itype + v6) == ITYPE_GOLD )
-					{
-						v7 = *(int *)((char *)&plr[0].SpdList[0]._ivalue + v6);
-						if ( v3 >= v7 )
-						{
-							v3 -= v7;
-							RemoveSpdBarItem(v1, ia);
-							SetPlrHandItem(&plr[v2].HoldItem, IDI_GOLD);
-							GetGoldSeed(v1, &plr[v2].HoldItem);
-							SetPlrHandGoldCurs(&plr[v2].HoldItem);
-							plr[v2].HoldItem._ivalue = *(int *)((char *)&plr[0].SpdList[0]._ivalue + v15);
-							PlrDeadItem(v1, &plr[v2].HoldItem, 0, 0);
-							ia = -1;
-						}
-						else
-						{
-							*(int *)((char *)&plr[0].SpdList[0]._ivalue + v6) = v7 - v3;
-							SetSpdbarGoldCurs(v1, ia);
-							SetPlrHandItem(&plr[v2].HoldItem, IDI_GOLD);
-							GetGoldSeed(v1, &plr[v2].HoldItem);
-							SetPlrHandGoldCurs(&plr[v2].HoldItem);
-							plr[v2].HoldItem._ivalue = v3;
-							v3 = 0;
-							PlrDeadItem(v1, &plr[v2].HoldItem, 0, 0);
-						}
-					}
-					++ia;
+	}
+	if ( hGold > 0 ) {
+		for ( i = 0; i < MAXBELTITEMS && hGold > 0; i++ ) {
+			if ( plr[pnum].SpdList[i]._itype == ITYPE_GOLD ) {
+				if ( hGold < plr[pnum].SpdList[i]._ivalue ) {
+					plr[pnum].SpdList[i]._ivalue -= hGold;
+					SetSpdbarGoldCurs(pnum, i);
+					SetPlrHandItem(&plr[pnum].HoldItem, IDI_GOLD);
+					GetGoldSeed(pnum, &plr[pnum].HoldItem);
+					SetPlrHandGoldCurs(&plr[pnum].HoldItem);
+					plr[pnum].HoldItem._ivalue = hGold;
+					PlrDeadItem(pnum, &plr[pnum].HoldItem, 0, 0);
+					hGold = 0;
+				} else {
+					hGold -= plr[pnum].SpdList[i]._ivalue;
+					RemoveSpdBarItem(pnum, i);
+					SetPlrHandItem(&plr[pnum].HoldItem, IDI_GOLD);
+					GetGoldSeed(pnum, &plr[pnum].HoldItem);
+					SetPlrHandGoldCurs(&plr[pnum].HoldItem);
+					plr[pnum].HoldItem._ivalue = plr[pnum].SpdList[i]._ivalue;
+					PlrDeadItem(pnum, &plr[pnum].HoldItem, 0, 0);
+					i = -1;
 				}
-				while ( ia < 8 );
 			}
-			break;
 		}
 	}
-	v8 = 0;
 	drawpanflag = 255;

```

Comment:
We should probably define™ an enum for this too, but that can be done in follow-up PRs.

Hehe, couldn't resist :)

Reply:
Almost dropped of my char when I got this message on my phone :D

Reply:
:P